v2/emailpassword/migration/account-creation/ep-migration-without-password-hash.mdx

```diff
@@ -0,0 +1,449 @@
+                        if (legacyUserInfo) {
+                            // Update the user's password with the correct password
+                            EmailPassword.updateEmailOrPassword({
+                                userId: supertokensUser.id,
+                                password: input.password
+                            })
+
+                            // Update the user's metadata to remove the isUsingTemporaryPassword flag
+                            UserMetadata.updateUserMetadata(supertokensUser.id, { isUsingTemporaryPassword: null })
+
+                            return {
+                                status: "OK",
+                                user: supertokensUser
+                            }
+                        }
+                        return {
+                            status: "WRONG_CREDENTIALS_ERROR"
+                        }
+                    }
+                    // highlight-end
+
+                    return originalImplementation.signIn(input)
+                }
+            }
+        }
+    }
+})
+
+
+async function validateAndGetUserInfoFromExternalProvider(email: string, password: string): Promise<{
+    user_id: string,
+    isEmailVerified: boolean
+} | undefined> {
+    // TODO: Validate the input credentials against the external authentication provider. If the credentials are valid return the user info.
+    return undefined
+}
+```
+
+</TabItem>
+</BackendSDKTabs>
+
+The code above adds the following changes to the `signIn` function:
+- Adds an additional check where if a user exists in SuperTokens, we check if they have the `isUsingTemporaryPassword` flag set in their metadata.
+- If the flag exists, we check if the input credentials are valid in the external provider. If they are we update the account with the new password and continue the login flow.
+- If the input credentials are invalid in the external provider, we return a `WRONG_CREDENTIALS_ERROR`.
+
+
+## User migration edge cases that are addressed
+This stratergy takes into account the following edge cases to ensure a smooth migration experience:
+- User forgets their passwords and tries the password reset flow
+  - In this situation, the regular password reset flow will not work since password resets require an existing account whose email the password reset link would be sent to.
+  - The changes proposed in [Step 3](#step-3-create-a-supertokens-account-for-users-who-have-an-account-with-the-external-provider-but-have-forgotten-their-password) and [Step 4](#step-4-remove-the-isusingtemporarypassword-flag-on-successful-password-reset) resolve the this edge case.  
+- User starts the password reset flow but they remember their password and try to login with the valid password.
+  - In this scenario if the user starts the password reset flow a SuperTokens account with a temporary password is created. Instead of completing the password reset flow they remember their password and try to sign in.
```

Comment:
```suggestion
  - In this scenario if the user starts the password reset flow, a new SuperTokens account with a temporary password is created. Instead of completing the password reset flow they remember their password and try to sign in.
```